ferret/explainers/shap.py

```diff
@@ -40,6 +40,8 @@ def compute_feature_importance(
         target_token_pos_idx = self.helper._check_target_token(text, target_token)
         text = self.helper._check_sample(text)
 
+        kwargs.pop('target_option', None)
```

Comment:
from here untile the end of the method: it is crucial to add some comments within the code to make it more readable.

Reply:
You are right, I added a few comments in the parts that seemed less readable.